src/cosmos/CosmosEngine.ts

```diff
@@ -410,14 +410,24 @@ export class CosmosEngine extends CurrencyEngine<
         mainnetBal?.amount ?? '0'
       )
 
-      this.enabledTokenIds.forEach(tokenId => {
+      const detectedTokenIds: string[] = []
+      Object.keys(this.allTokensMap).forEach(tokenId => {
         const token = this.allTokensMap[tokenId]
         const tokenBal = balances.find(
           bal => bal.denom === token.networkLocation?.contractAddress
         )
-        this.updateBalance(token.currencyCode, tokenBal?.amount ?? '0')
+        const balance = tokenBal?.amount ?? '0'
+        this.updateBalance(token.currencyCode, balance)
```

Comment:
Would we want to guard the `updateBalance` call with `this.enabledTokenIds.includes(tokenId)`, or does it not matter?

Reply:
As we discussed IRL, updateBalance doesn't fire the expensive callback unless the balance changes so there's it's not necessary to guard it.